hydromt/models/model.py

```diff
@@ -54,22 +47,11 @@ class Model(object, metaclass=ABCMeta):
     Inherit from this class to pre-define mandatory components in the model.
     """
 
-    _DATADIR = ""  # path to the model data folder
     _NAME: str = "modelname"
-    _MAPS = {"<general_hydromt_name>": "<model_name>"}
-    _FOLDERS = [""]
-    _TMP_DATA_DIR = None
     # supported model version should be filled by the plugins
     # e.g. _MODEL_VERSION = ">=1.0, <1.1"
     _MODEL_VERSION = None
 
-    _API = {
-        "maps": XArrayDict,
-        "forcing": XArrayDict,
-        "results": XArrayDict,
-        "states": XArrayDict,
-    }
```

Comment:
Do we completely delete? Or leave the dict empty in core? Maybe these API tests are not so needed anymore because we use pydantic and do better type checking in core?

Reply:
Indeed, these are not necessary any more as they are taken care of by the component architecture.